kolibri/core/tasks/job.py

```diff
@@ -156,9 +172,43 @@ def __getstate__(self):
             "args",
             "kwargs",
             "func",
-            "result",
         ]
-        return {key: self.__dict__[key] for key in keys if key in self.__dict__}
+        working_dictionary = {key: self.__dict__[key] for key in keys if key in self.__dict__}
+
+        if self.exception is not None:
+            working_dictionary["exception_type"] = type(self.exception).__name__
+            if self.exception.args[0] and isinstance(self.exception.args[0], str):
+                working_dictionary["exception_message"] = self.exception.args[0]
+
```

Comment:
Following from my above recommendation:

We should be setting `self.exception` to `self.exception.__name__`.

```
if self.exception is not None:
    self.exception = self.exception.__name__
```